nat-lab/tests/utils/batching.py

```diff
@@ -0,0 +1,161 @@
+# There's a dillemma in how to best capture the traffic. Capturing on the node itself
+# provides insights into the packets themselves coming to and from the tunnel device,
+# however capturing locally doesn't mean the packet actually left the machine.
+# An alternative is to capture on gateways which guarantees the packet arrival
+# however there's no way to easily pry open the packets and observe their insides.
+import asyncio
+import math
+import os
+import pytest
+import subprocess
+import typing
+from scapy.all import PcapReader  # type: ignore
+from scapy.layers.inet import IP, ICMP  # type: ignore
+from typing import Any, Callable, List, Tuple
+
+
+class Histogram:
```

Comment:
This looks more like a HistogramFactory tbh, as sth which could be called really a histogram is generated by `get` method, here you have just a list of floats

Reply:
That's actually a good call. No need for Histogram to be anything else except a free function. I've refactored it out into a such function now.

Reply:
I refactored code a bit around so now `histogram` is only mentioned when it's generated from data. I think it's leaner that way and became easier to follow